dhis-2/dhis-services/dhis-service-analytics/src/test/java/org/hisp/dhis/analytics/orgunit/OrgUnitAnalyticsServiceTest.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2004-2020, University of Oslo
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * Neither the name of the HISP project nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.hisp.dhis.analytics.orgunit;
+
+import static org.hisp.dhis.analytics.util.AnalyticsTestUtils.assertIllegalQueryEx;
+
+import org.hisp.dhis.DhisSpringTest;
+import org.hisp.dhis.feedback.ErrorCode;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import com.google.common.collect.Lists;
+
+/**
+ * @author Lars Helge Overland
+ */
+public class OrgUnitAnalyticsServiceTest
+    extends DhisSpringTest
+{
+    @Autowired
+    private OrgUnitAnalyticsService subject;
+
+    @Test
+    public void testValidateNoOrgUnits()
+    {
+        OrgUnitQueryParams params = new OrgUnitQueryParams.Builder()
+            .withOrgUnitGroupSets( Lists.newArrayList( createOrganisationUnitGroupSet( 'A' ) ) )
+            .build();
+
+        assertIllegalQueryEx( p -> subject.validate( p ), params, ErrorCode.E7300 );
```

Comment:
Minor: you can also consider using the `ExpectedException` Rule, for asserting on exceptions, like so (or use an utility method that operates on the `ExpectedException` object):

```
@Rule
    public ExpectedException exception = ExpectedException.none();

    @Test
    public void testValidateNoOrgUnits()
    {
        OrgUnitQueryParams params = new OrgUnitQueryParams.Builder()
            .withOrgUnitGroupSets( Lists.newArrayList( createOrganisationUnitGroupSet( 'A' ) ) )
            .build();

        exception.expectMessage( "At least one organisation unit must be specified" );
        exception.expect( IllegalQueryException.class );
        exception.reportMissingExceptionWithMessage( "Expected exception" );
        exception.expect( Matchers.hasProperty( "errorCode", CoreMatchers.is( ErrorCode.E7300 ) ) );

        subject.validate( params );
    }
```

Reply:
Great, I was aware of `ExceptedException` but not:

`exception.expect( Matchers.hasProperty( "errorCode", CoreMatchers.is( ErrorCode.E7300 ) ) );`

Thanks!

Reply:
Rewrote it now using Junit/Hamcrest.